water_mark/water_mark.py

```diff
@@ -0,0 +1,172 @@
+        vh = data_list[i]
+        data[sar_name] = (vv, vh)
+    return data
+
+
+def make_output_dir(out_dir: str, data_dict: Dict[str, Tuple[str, str]]) -> None:
+    if os.path.exists(out_dir):
+        shutil.rmtree(out_dir)
+    os.mkdir(out_dir)
+    for sar in data_dict:
+        os.mkdir(os.path.join(out_dir, sar))
+
+
+def delete_junk(target_dir):
+    for f_name in os.listdir(directory):
+        if ('.shp') in f_name:
+            os.remove(os.path.join('inputs', f_name))
+        if ('.shx') in f_name:
+            os.remove(os.path.join('inputs', f_name))
+        if ('.prj') in f_name:
+            os.remove(os.path.join('inputs', f_name))
+        if ('.dbf') in f_name:
+            os.remove(os.path.join('inputs', f_name))
+
+
+def copy_vv_vh_to_inputs(out_dir, data_dict):
+    for sar, vv_vh_band in data_dict.items():
+        shutil.copy(
+            os.path.join('inputs', vv_vh_band[0]),
+            os.path.join(out_dir, sar)
+        )
+        shutil.copy(
+            os.path.join('inputs', vv_vh_band[1]),
+            os.path.join(out_dir, sar)
+        )
+
+
+def make_masks(out_dir, data_dict):
+    count = 0
+    for sar, vv_vh_band in data_dict.items():
+        renamePath = os.path.join(out_dir, sar)
+        print(f"{count} masks made")
+        idw_main(
+            os.path.join('inputs', vv_vh_band[0]),
+            os.path.join('inputs', vv_vh_band[1])
+        )
+        count += 1
+        os.rename(
+            'mask-0.tif',
+            os.path.join(renamePath, f"Mask_{sar}.tif")
+        )
+
+
+def tile(out_dir, tif_name, sar, mxm_tile_size, isMask):
+    label = 'temp'
+    if isMask:
+        label = 'Mask'
+    else:
+        label = 'Image'
+    tif = os.path.join(out_dir, sar, tif_name)
+    tifData = gdal.Open(tif)
+    xStep, yStep = mxm_tile_size, mxm_tile_size
+    xSize, ySize = tifData.RasterXSize, tifData.RasterYSize
+    count = 0
+    for x in range(0, xSize, xStep):
+        for y in range(0, ySize, yStep):
+            fileName = f"{label}_{tif_name[:-4]}_{count}.tif"
+            output_file = os.path.join(out_dir, sar, fileName)
+            input_file = tif
+            gdal.Translate(
+                output_file,
+                input_file,
+                srcWin=[x, y, xStep, yStep],
+                format="GTiff"
+            )
+            count += 1
+
+
+def tile_vv_vh_mask(out_dir, mxm_tile_size):
+    for sar in os.listdir(out_dir):
+        for tif_name in os.listdir(os.path.join(out_dir, sar)):
+            if tif_name.endswith('VV.tif'):
+                tile(out_dir, tif_name, sar, mxm_tile_size, False)
+            elif tif_name.endswith('VH.tif'):
+                tile(out_dir, tif_name, sar, mxm_tile_size, False)
+            elif tif_name.startswith('Mask_'):
+                tif_name = tif_name[5:]  # remove leading 'Mask_'
+                tile(out_dir, tif_name, sar, mxm_tile_size, True)
+
+
+def main():
+    parser = argparse.ArgumentParser()
```

Comment:
I sent a pull request to your fork updating this code to play more friendly with other portions of the code. If you could review and merge it before we merge this that would be perfect.